gobblin-utility/src/main/java/org/apache/gobblin/util/LdapUtils.java

```diff
@@ -87,7 +94,11 @@ private DirContext getDirContext(String username, String password) throws Naming
     env.put(Context.SECURITY_PRINCIPAL, username);
     env.put(Context.SECURITY_CREDENTIALS, password);
 
-    env.put("java.naming.ldap.factory.socket", TrustManagerSocketFactory.class.getCanonicalName());
+    if (_ldapUseSecureTrustManager) {
+      env.put("java.naming.ldap.factory.socket", TrustManagerSocketFactory.class.getCanonicalName());
```

Comment:
I think you mixed up the branches here as this one should be the secure one and the other one the non-secure one.

Reply:
yes, good catch. fixing it